pkg/vsphere/disk/util.go

```diff
@@ -260,5 +291,22 @@ func findDisk(op trace.Operation, vm *vm.VirtualMachine, name string) (*types.Vi
 		return nil, errors.Errorf("Too many disks match name: %s", name)
 	}
 
-	return candidates[0].(*types.VirtualDisk), nil
+	return candidates[0], nil
+}
+
+func findAllDisks(op trace.Operation, vm *vm.VirtualMachine) ([]*types.VirtualDisk, error) {
+	defer trace.End(trace.Begin(vm.String()))
+
+	op.Debugf("Looking for all attached disks")
+
+	candidates, err := findDisk(op, vm, func(diskName string) bool {
```

Comment:
why candidates? I think naming it as "disks"  would be better. These are no longer candidates since you pre-filtered them already.